# Контрольная работа

08.11.2021

Выполнил студент ПИ19-3 Ковалев А.И.

In [229]:
import numpy_financial as npf
import sympy as sp

import bond
import intrest

## Задача 1

In [230]:
start_a = 1000
end_a = 1110
i_a = sp.Symbol('i')
t_a = sp.Symbol('t')

start_b = 500
i_b = 3 * i_a
t_b = 2 * t_a

Формула простых процентов:

In [231]:
intrest.simple_interest(*sp.symbols('S S_0 i t'))

Eq(S, S_0*(i*t + 1))

In [232]:
f1 = intrest.simple_interest(end_a, start_a, i_a, t_a)
f1

Eq(1110, 1000*i*t + 1000)

In [233]:
it = sp.solve(f1, [i_a * t_a])[0]
it

11/100

In [234]:
f2 = intrest.simple_interest(sp.Symbol('S'), start_b, i_b, t_b)
f2

Eq(S, 3000*i*t + 500)

In [235]:
end_b = f2.subs(i_a * t_a, it)
end_b

Eq(S, 830)

In [236]:
end_b_value = sp.solve(end_b)[0]
end_b_value

830

**Ответ: 830**

## Задача 2

In [237]:
i = 0.155
s_0 = 1_000_000
t = 5

Формула сложных процентов:

In [238]:
intrest.compound_interest(*sp.symbols('S S_0 i t'))

Eq(S, S_0*(i + 1)**t)

In [239]:
s = intrest.compound_interest(sp.Symbol('S'), s_0, i, t)
s

Eq(S, 2055464.21922188)

In [240]:
s_value = sp.solve(s)[0]
s_value

2055464.21922188

**Ответ: 2055464.22**

## Задача 3

Идея решения задачи:

$ NPV = \sum_k{S_{t_k} (1 + i)^{-t_k}} $

In [241]:
f_a = [-1000, 300, 200, 100, 100, 800]
f_b = [-1000, 800, 100, 100, 200, 300]
i = 0.1

In [242]:
npv_a = npf.npv(i, f_a)
npv_a

78.18641299960859

In [243]:
npv_b = npf.npv(i, f_b)
npv_b

207.92792345281913

**Ответ: инвестиционный проект B лучше (207.93 > 78.19)**

## Задача 4

Идея решения задачи:

$ \frac{\Delta P}{P} = -D^* \Delta \gamma $
<br>
$ D^* = \frac{D}{1 + \gamma} $

In [244]:
p_1 = 2000
y_1 = 0.06
y_2 = 0.07
duration = 8

In [245]:
m_duration = duration / (1 + y_1)
m_duration

7.547169811320754

In [246]:
delta_p = sp.Eq(sp.Symbol('\Delta P') / sp.Symbol('P'), -m_duration * (y_2 - y_1)).subs(sp.Symbol('P'), p_1)
delta_p

Eq(\Delta P/2000, -0.0754716981132076)

In [247]:
delta_p_value = sp.solve(delta_p)[0]
delta_p_value

-150.943396226415

In [248]:
p2 = p_1 + delta_p_value
p2

1849.05660377358

**Ответ: 1849.057**

## Задача 5

Идея решения задачи:

$ NPV = \frac{C}{i} (1 - \frac{1}{(1 + i)^t}) +  \frac{F}{(1 + i)^t}$

In [249]:
f = 1000
t = 10
c = 0.1
i = 0.08

In [250]:
npv = bond.npv(sp.Symbol('NPV'), f, c * f, i, t)
npv

Eq(NPV, 1134.20162797883)

In [251]:
npv_value = sp.solve(npv)[0]
npv_value

1134.20162797883

**Ответ: 1134.202**
